src/eth1/impl/ethers.ts

```diff
@@ -50,12 +51,23 @@ export class EthersEth1Notifier extends EventEmitter implements Eth1Notifier {
     if(!this.contract) {
       await this.initContract();
     }
-    this.provider.on('block', this.processBlockHeadUpdate.bind(this));
-    this.contract.on('Deposit', this.processDepositLog.bind(this));
-    this.contract.on('Eth2Genesis', this.processEth2GenesisLog.bind(this));
-    logger.info(
-      `Started listening on eth1 events on chain ${(await this.provider.getNetwork()).chainId}`
-    );
+    if(await this.isAfterEth2Genesis()) {
+      logger.info('Eth2Genesis event exits, started listening on eth1 block updates');
+      this.provider.on('block', this.processBlockHeadUpdate.bind(this));
+    } else {
+      const pastDeposits = await this.getPastDepositEvents();
+      await Promise.all(pastDeposits.map((pastDeposit) => {
+        return this.db.setGenesisDeposit(pastDeposit);
+      }));
```

Comment:
I think we we're missing deposits between genesis and now that we'll need to calculate proofs?

Reply:
This is case when genesis has not occured, we just have deposit logs till now. But now that you mentioned, there is a bug toBlock in filter should default to latest I guess

Reply:
I think @mpetrunic is correct, genesis hasn't fired